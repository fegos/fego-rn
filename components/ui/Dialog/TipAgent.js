/**
 * 提示框容器
 */
import React from 'react';
import { Text, ScrollView } from 'react-native';
import { UIComponent } from '../../common';

import Dialog from './index';

export default class TipAgent extends UIComponent {
  constructor(props) {
    super(props);
    this.state = {
      visible: true,
    };
  }
  componentWillUnmount() {
    this.close = () => { };
  }
  close(fn) {
    this.setState({
      visible: false,
    }, fn);
  }
  _crtContent(style) {
    const { content, title } = this.props;
    if (typeof content === 'string') {
      return <Text style={[style.content, title ? style.contentHasTitle : null]}>{content}</Text>;
    }
    return content;
  }
  render() {
    const { style } = this;
    const {
      title, actions, onAnimationEnd, type,
    } = this.props;
    const setAniEndFn = (endFn) => {
      this.endFn = endFn;
    };
    const _onAniEnd = (visible) => {
      if (onAnimationEnd) {
        onAnimationEnd(visible);
      }
      // 动画关闭时回调
      if (this.endFn && !visible) {
        this.endFn(visible);
      }
    };
    const footer = actions.map((button) => {
      const orginPress = button.onPress || function temp() { };
      button.onPress = () => {
        const res = orginPress(button, setAniEndFn);
        // 若返回false则不关闭
        if (res !== false) {
          this.close();
          return;
        }
        // 若返回promise则resolve时才关闭
        if (res && res.then) {
          res.then(() => {
            this.close();
          });
        }
      };
      return button;
    });
    const content = this._crtContent(style);
    return (
      <Dialog
        title={title}
        type={type}
        visible={this.state.visible}
        onClose={() => this.close}
        footer={footer}
        onAnimationEnd={_onAniEnd}
      >
        <ScrollView style={style.container} >
          {content}
        </ScrollView>
      </Dialog>
    );
  }
}
TipAgent.baseStyle = {
  container: {

  },
  content: {
    fontSize: 16,
    lineHeight: 22,
    color: '#030303',
    textAlign: 'center',
  },
  contentHasTitle: {
    color: '#888',
  },
};
