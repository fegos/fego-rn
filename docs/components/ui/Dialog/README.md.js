webpackJsonp([30],{472:function(n,s){n.exports={content:["article",["h1","Dialog\u5bf9\u8bdd\u6846"],["h2","\u793a\u4f8b"],["pre",{lang:"js",highlighted:'Dialog<span class="token punctuation">.</span><span class="token function">alert</span><span class="token punctuation">(</span><span class="token string">\'\u5185\u5bb9\'</span><span class="token punctuation">)</span>\n\nDialog<span class="token punctuation">.</span><span class="token function">alert</span><span class="token punctuation">(</span><span class="token string">\'\u6807\u9898\'</span><span class="token punctuation">,</span> <span class="token string">\'\u5185\u5bb9\'</span><span class="token punctuation">)</span>\n\nDialog<span class="token punctuation">.</span><span class="token function">alert</span><span class="token punctuation">(</span><span class="token string">\'\u5185\u5bb9\'</span><span class="token punctuation">,</span> <span class="token punctuation">(</span>btn<span class="token punctuation">)</span><span class="token operator">=</span><span class="token operator">></span><span class="token punctuation">{</span><span class="token punctuation">}</span><span class="token punctuation">)</span>\n\nDialog<span class="token punctuation">.</span><span class="token function">confirm</span><span class="token punctuation">(</span><span class="token string">\'\u4f60\u5f88\u5e05\u5417\uff1f\'</span><span class="token punctuation">,</span> <span class="token punctuation">(</span>btn<span class="token punctuation">)</span><span class="token operator">=</span><span class="token operator">></span><span class="token punctuation">{</span>\n\t<span class="token keyword">if</span><span class="token punctuation">(</span>btn<span class="token punctuation">.</span>type <span class="token operator">===</span> <span class="token string">\'yes\'</span><span class="token punctuation">)</span><span class="token punctuation">{</span>\n\t\tDialog<span class="token punctuation">.</span><span class="token function">alert</span><span class="token punctuation">(</span><span class="token string">\'\u4f60\u5f88\u81ea\u4fe1\u561b\uff01\'</span><span class="token punctuation">)</span>\n\t<span class="token punctuation">}</span><span class="token keyword">else</span><span class="token punctuation">{</span>\n\t\tDialog<span class="token punctuation">.</span><span class="token function">alert</span><span class="token punctuation">(</span><span class="token string">\'\u8fd9\u5c31\u5bf9\u4e86\uff01\'</span><span class="token punctuation">)</span>\n\t<span class="token punctuation">}</span>\n<span class="token punctuation">}</span><span class="token punctuation">)</span>\n\nDialog<span class="token punctuation">.</span><span class="token function">confirm</span><span class="token punctuation">(</span><span class="token operator">&lt;</span>Text style<span class="token operator">=</span><span class="token punctuation">{</span><span class="token punctuation">{</span>color<span class="token punctuation">:</span> <span class="token string">\'#338811\'</span><span class="token punctuation">}</span><span class="token punctuation">}</span><span class="token operator">></span>\u8fd9\u7a97\u53e3\u70b9\u786e\u5b9a\u662f\u6ca1\u7528\u7684<span class="token operator">&lt;</span><span class="token operator">/</span>Text<span class="token operator">></span><span class="token punctuation">,</span> <span class="token punctuation">(</span>btn<span class="token punctuation">)</span><span class="token operator">=</span><span class="token operator">></span><span class="token punctuation">{</span>\n\t<span class="token keyword">if</span><span class="token punctuation">(</span>btn<span class="token punctuation">.</span>type <span class="token operator">===</span> <span class="token string">\'yes\'</span><span class="token punctuation">)</span><span class="token punctuation">{</span>\n\t\t<span class="token keyword">return</span> <span class="token boolean">false</span>\n\t<span class="token punctuation">}</span>\n<span class="token punctuation">}</span><span class="token punctuation">)</span>\n\nDialog<span class="token punctuation">.</span><span class="token function">confirm</span><span class="token punctuation">(</span><span class="token string">\'\u52a8\u753b\u7ed3\u675f\u56de\u8c03\'</span><span class="token punctuation">,</span> <span class="token punctuation">(</span>btn<span class="token punctuation">,</span> aniEnd<span class="token punctuation">)</span><span class="token operator">=</span><span class="token operator">></span><span class="token punctuation">{</span>\n\tconsole<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">\'\u6309\u94ae\u88ab\u70b9\u51fb\uff0c\u4f46\u52a8\u753b\u672a\u7ed3\u675f\'</span><span class="token punctuation">)</span>\n\t<span class="token function">aniEnd</span><span class="token punctuation">(</span><span class="token punctuation">(</span><span class="token punctuation">)</span><span class="token operator">=</span><span class="token operator">></span><span class="token punctuation">{</span>\n\t\tconsole<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">\'\u52a8\u753b\u7ed3\u675f\u540e\u624d\u6267\u884c\'</span><span class="token punctuation">)</span>\n\t<span class="token punctuation">}</span><span class="token punctuation">)</span>\n<span class="token punctuation">}</span><span class="token punctuation">)</span>\n\nDialog<span class="token punctuation">.</span><span class="token function">tip</span><span class="token punctuation">(</span><span class="token string">\'\u5220\u9664\'</span><span class="token punctuation">,</span> <span class="token string">\'\u786e\u5b9a\u5220\u9664\u4e48???\'</span><span class="token punctuation">,</span> <span class="token punctuation">[</span>\n\t<span class="token punctuation">{</span> text<span class="token punctuation">:</span> <span class="token string">\'\u4e0d\u597d\'</span><span class="token punctuation">,</span> onPress<span class="token punctuation">:</span> <span class="token punctuation">(</span><span class="token punctuation">)</span> <span class="token operator">=</span><span class="token operator">></span> console<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">\'cancel\'</span><span class="token punctuation">)</span> <span class="token punctuation">}</span><span class="token punctuation">,</span>\n\t<span class="token punctuation">{</span> text<span class="token punctuation">:</span> <span class="token string">\'\u53ef\u4ee5\'</span><span class="token punctuation">,</span> onPress<span class="token punctuation">:</span> <span class="token punctuation">(</span><span class="token punctuation">)</span> <span class="token operator">=</span><span class="token operator">></span> console<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">\'ok\'</span><span class="token punctuation">)</span><span class="token punctuation">,</span> style<span class="token punctuation">:</span> <span class="token punctuation">{</span> fontWeight<span class="token punctuation">:</span> <span class="token string">\'bold\'</span> <span class="token punctuation">}</span> <span class="token punctuation">}</span><span class="token punctuation">,</span>\n<span class="token punctuation">]</span><span class="token punctuation">)</span>'},["code","Dialog.alert('\u5185\u5bb9')\n\nDialog.alert('\u6807\u9898', '\u5185\u5bb9')\n\nDialog.alert('\u5185\u5bb9', (btn)=>{})\n\nDialog.confirm('\u4f60\u5f88\u5e05\u5417\uff1f', (btn)=>{\n\tif(btn.type === 'yes'){\n\t\tDialog.alert('\u4f60\u5f88\u81ea\u4fe1\u561b\uff01')\n\t}else{\n\t\tDialog.alert('\u8fd9\u5c31\u5bf9\u4e86\uff01')\n\t}\n})\n\nDialog.confirm(<Text style={{color: '#338811'}}>\u8fd9\u7a97\u53e3\u70b9\u786e\u5b9a\u662f\u6ca1\u7528\u7684</Text>, (btn)=>{\n\tif(btn.type === 'yes'){\n\t\treturn false\n\t}\n})\n\nDialog.confirm('\u52a8\u753b\u7ed3\u675f\u56de\u8c03', (btn, aniEnd)=>{\n\tconsole.log('\u6309\u94ae\u88ab\u70b9\u51fb\uff0c\u4f46\u52a8\u753b\u672a\u7ed3\u675f')\n\taniEnd(()=>{\n\t\tconsole.log('\u52a8\u753b\u7ed3\u675f\u540e\u624d\u6267\u884c')\n\t})\n})\n\nDialog.tip('\u5220\u9664', '\u786e\u5b9a\u5220\u9664\u4e48???', [\n\t{ text: '\u4e0d\u597d', onPress: () => console.log('cancel') },\n\t{ text: '\u53ef\u4ee5', onPress: () => console.log('ok'), style: { fontWeight: 'bold' } },\n])"]]],meta:{title:"Dialog",subTitle:"\u5bf9\u8bdd\u6846",filename:"components/ui/Dialog/README.md"},toc:["ul",["li",["a",{className:"bisheng-toc-h1",href:"#Dialog\u5bf9\u8bdd\u6846",title:"Dialog\u5bf9\u8bdd\u6846"},"Dialog\u5bf9\u8bdd\u6846"]],["li",["a",{className:"bisheng-toc-h2",href:"#\u793a\u4f8b",title:"\u793a\u4f8b"},"\u793a\u4f8b"]]]}}});