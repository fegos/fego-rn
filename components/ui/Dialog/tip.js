/**
 * 简易的提示
 * 基于模态实现，同一时刻只能创建一个提示框
 */
import React from 'react';
import AppContainer from '../AppContainer';
import TipAgent from './TipAgent';

const defaultOption = {
  alert: {
    // title: '提示',
    actions: [{ text: '确定', type: 'yes' }],
  },
  confirm: {
    // title: '请确认',
    actions: [{ text: '取消', type: 'no', style: { color: '#858DA0' } }, { text: '确定', type: 'yes' }],
  },
};
const topViewDialogKey = 'modal.Dialog.tip';
let curTip = null;
const onAnimationEnd = (visible) => {
  if (!visible) {
    curTip = null;
    AppContainer.remove(topViewDialogKey);
  }
};

function creatTip(title, content, actions, type) {
  actions = actions || defaultOption.alert.actions;
  AppContainer.add(<TipAgent
    key={topViewDialogKey}
    ref={(el) => { curTip = el; }}
    title={title}
    type={type}
    content={content}
    actions={actions}
    onAnimationEnd={onAnimationEnd}
  />);
}

/**
 * 基础的提示框（唯一不可叠加）
 * @param {String} title
 * @param {String/Element} content
 * @param {Array} actions
 */
function tip(title, content, actions, type) {
  if (curTip) {
    curTip.close(() => {
      setTimeout(() => {
        AppContainer.remove(topViewDialogKey);
        setTimeout(() => {
          creatTip(title, content, actions);
        }, 0);
      }, 200);
    });
  } else {
    creatTip(title, content, actions, type);
  }
}

/**
 * 提醒
 * Dialog.alert('内容')
 * Dialog.alert('内容', callback)
 * Dialog.alert('标题', '内容')
 * Dialog.alert('标题', '内容', callback)
 * @param {String} title
 * @param {String/Element} content
 */
function alert(title, content, callback, type) {
  const opt = defaultOption.alert;
  const argLen = arguments.length;
  if (argLen === 1) {
    content = title;
    title = null;
  }
  if (argLen === 2) {
    if (typeof content === 'function') {
      callback = content;
      content = title;
      title = null;
    }
  }
  title = title || opt.title;
  if (opt.actions[0]) {
    opt.actions[0].onPress = callback;
  }
  tip(title, content, opt.actions, type);
}
/**
 * 确认
 * Dialog.confirm('内容')
 * Dialog.confirm('内容', callback)
 * Dialog.confirm('标题', '内容')
 * Dialog.confirm('标题', '内容', callback)
 * @param {String} title
 * @param {String/Element} content
 */
function confirm(title, content, callback, type) {
  const opt = defaultOption.confirm;
  const argLen = arguments.length;
  if (argLen === 1) {
    content = title;
    title = null;
  }
  if (argLen === 2) {
    if (typeof content === 'function') {
      callback = content;
      content = title;
      title = null;
    }
  }
  title = title || opt.title;
  if (opt.actions[0]) {
    opt.actions[0].onPress = callback;
  }
  if (opt.actions[1]) {
    opt.actions[1].onPress = callback;
  }
  tip(title, content, opt.actions, type);
}
/**
 * TipAgent容器，可设置基础样式
 */
tip.TipAgent = TipAgent;
/**
 * 设置默认配置
 */
tip.setDefaultOption = function setOptions(option) {
  Object.assign(defaultOption, option);
};
export {
  tip,
  alert,
  confirm,
};
export default tip;
