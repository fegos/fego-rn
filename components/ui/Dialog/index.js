/**
 * 对话框
 * 含标题，按钮区等
 */
import React from 'react';
import PropTypes from 'prop-types';
import {
  View,
  Text,
  TouchableHighlight,
  Dimensions,
  StyleSheet,
  TouchableWithoutFeedback,
} from 'react-native';
import { UIComponent } from 'common';

import AnimateModal from '../AnimateModal';
import { tip, alert, confirm } from './tip';


const { maxHeight } = StyleSheet.create({
  maxHeight: {
    maxHeight: Dimensions.get('window').height,
  },
});

export default class Dialog extends UIComponent {
  static defaultProps = {
    visible: false,
    closable: false,
    maskClosable: false,
    animationType: 'fade',
    footer: [],
    fullScreen: false,
    animateWhenMount: true,
    btnUnderlayColor: '#EEE',
    footerBtnStyleMap: {
      cancel: { color: '#666' },
      confirm: { color: '#337ab7' },
    },
    onClose: () => { },
  }

  static propTypes = {
    // 是否可见
    visible: PropTypes.bool,
    // 是否可关闭
    closable: PropTypes.bool,
    // 蒙层是否可关闭
    maskClosable: PropTypes.bool,
    // 动画类型
    animationType: PropTypes.string,
    // 关闭
    onClose: PropTypes.func,
    // 底部按钮数组
    footer: PropTypes.arrayOf(PropTypes.any),
    // 是否全屏
    fullScreen: PropTypes.bool,
    // 是否使用动画
    animateWhenMount: PropTypes.bool,
    // 按钮点击颜色
    btnUnderlayColor: PropTypes.string,
    // 按钮类型
    footerBtnStyleMap: PropTypes.objectOf(PropTypes.any),
  }

  /**
   * 简易的提示功能
   */
  static tip = tip
  static alert = alert
  static confirm = confirm
  static autoStyleSheet = false
  _maskClose = () => {
    if (this.props.maskClosable && this.props.onClose) {
      this.props.onClose();
    }
  }

  render() {
    const styles = this.style;
    const {
      children, animateWhenMount, fullScreen, visible, onAnimationEnd, animationType, animationDuration,
    } = this.props;
    if (fullScreen) {
      return this._createFullScreenDialog(styles);
    }
    const headerEl = this._createHeader(styles);
    const footerEl = this._createFooter(styles);
    const closeEl = this._createClose(styles);
    return (
      <View style={styles.container}>
        <AnimateModal
          onClose={this._maskClose}
          animationType={animationType}
          animationDuration={animationDuration}
          style={styles.inner}
          contentStyle={styles.innerContent}
          visible={visible}
          onAnimationEnd={onAnimationEnd}
          animateWhenMount={animateWhenMount}
        >
          <View style={maxHeight} ref={(ref) => { this.root = ref; }}>
            {headerEl}
            <View style={styles.body}>{children}</View>
            {footerEl}
            {closeEl}
          </View>
        </AnimateModal>
      </View >
    );
  }

  _createHeader = (styles) => {
    const { title } = this.props;
    return title ? <Text style={[styles.header]}>{title}</Text> : null;
  }


  _createFooter = (styles) => {
    const {
      footer, onClose, btnUnderlayColor, footerBtnStyleMap,
    } = this.props;
    if (!footer || !footer.length) {
      return null;
    }
    let btnGroupStyle = styles.buttonGroupV;
    const borderRadius = styles.innerContent.borderRadius || 10;
    let horizontalFlex = {};
    let isHorizontal = false;
    if (footer && footer.length === 2) {
      isHorizontal = true;
      btnGroupStyle = styles.buttonGroupH;
      horizontalFlex = { flex: 1 };
    }
    const buttonWrapStyle = footer && footer.length === 2 ? styles.buttonWrapH : styles.buttonWrapV;
    const footerButtons = footer.map((button, i) => {
      let buttonStyle = {};
      if (button.style) {
        buttonStyle = button.style;
        if (typeof buttonStyle === 'string') {
          buttonStyle = footerBtnStyleMap[buttonStyle] || {};
        }
      }
      const noneBorder = footer && footer.length === 2 && i === 1 ? { borderRightWidth: 0 } : {};
      const onPressFn = () => {
        if (button.onPress) {
          button.onPress(button);
        }
        if (onClose) {
          onClose();
        }
      };
      let borderBottomLeftRadius;
      if (isHorizontal) {
        borderBottomLeftRadius = i === 0 ? borderRadius : 0;
      } else {
        borderBottomLeftRadius = i === footer.length - 1 ? borderRadius : 0;
      }
      const key = `th${i}`;
      return (
        <TouchableHighlight
          key={key}
          style={[horizontalFlex, {
            /* 修复安卓按钮按住高亮时，若容器有圆角，边角仍然溢出的bug */
            borderBottomLeftRadius,
            borderBottomRightRadius: i === footer.length - 1 ? borderRadius : 0,
          }]}
          underlayColor={btnUnderlayColor}
          onPress={onPressFn}
        >
          <View style={[buttonWrapStyle, noneBorder]}>
            <Text style={[styles.buttonText, buttonStyle]}>{button.text || ''}</Text>
          </View>
        </TouchableHighlight>
      );
    });
    return (
      <View style={[btnGroupStyle, styles.footer]}>
        {footerButtons}
      </View>
    );
  }

  _createClose = (styles) => {
    const { closable, onClose } = this.props;
    return closable ? (
      <View style={[styles.closeWrap]}>
        <TouchableWithoutFeedback onPress={onClose}>
          <View>
            <Text style={[styles.close]}>×</Text>
          </View>
        </TouchableWithoutFeedback>
      </View>
    ) : null;
  }

  _createFullScreenDialog = (styles) => {
    const {
      children, visible, animationType,
    } = this.props;
    let aniType = animationType;
    if (animationType === 'slide-up' || animationType === 'slide-down' || animationType === 'slide') {
      aniType = 'slide';
    }
    return (
      <View style={styles.container}>
        <AnimateModal
          visible={visible}
          animationType={aniType}
          onRequestClose={this._maskClose}
        >
          <View style={styles.body}>
            {children}
          </View>
        </AnimateModal>
      </View>
    );
  }
}

Dialog.baseStyle = {
  container: {
    zIndex: 1000,
  },
  // 内部样式 （即modal容器）
  inner: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  // 内部内容区样式 （即modal内容区）
  innerContent: {
    borderRadius: 10,
    width: 270,
    overflow: 'hidden',
  },
  footer: {
    bottom: 0,
    left: 0,
    right: 0,
  },
  header: {
    fontSize: 20,
    color: '#000',
    textAlign: 'center',
    paddingTop: 20,
    paddingHorizontal: 22,
  },
  body: {
    paddingTop: 20,
    paddingBottom: 20,
    paddingHorizontal: 22,
  },
  maskClosable: {
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    backgroundColor: 'transparent',
  },
  closeWrap: {
    position: 'absolute',
    top: 0,
    left: 15,
  },
  close: {
    fontSize: 40,
    fontWeight: '200',
    color: '#bcbcbc',
    lineHeight: 44,
  },
  // 水平按钮组
  buttonGroupH: {
    flexGrow: 1,
    flexDirection: 'row',
  },
  // 垂直按钮组
  buttonGroupV: {
    flexGrow: 1,
    flexDirection: 'column',
  },
  buttonWrapH: {
    flexGrow: 1,
    borderColor: '#EEE',
    borderTopWidth: StyleSheet.hairlineWidth,
    borderRightWidth: StyleSheet.hairlineWidth,
    paddingVertical: 11,
  },
  buttonWrapV: {
    flexGrow: 1,
    borderTopWidth: StyleSheet.hairlineWidth,
    borderColor: '#EEE',
    paddingVertical: 11,
  },
  buttonText: {
    textAlign: 'center',
    color: '#0076FF',
    fontSize: 17,
    backgroundColor: 'transparent',
  },
};
