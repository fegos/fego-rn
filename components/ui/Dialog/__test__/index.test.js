import React from 'react';
import Renderer from 'react-test-renderer';
import { View, Text } from 'react-native';
import Dialog from '../index';

jest.useFakeTimers();
describe('Dialog Tests', () => {
  let Comp;
  let handler;
  beforeEach(() => {
    handler = jest.fn();
    Comp = (
      <Dialog
        title="标题"
        visible
        animateAppear={false}
        onClose={handler}
      >
        <View>
          <Text>简单对话框</Text>Ï
        </View>
      </Dialog>
    );
  });

  it('renders correctly', () => {
    const tree = Renderer.create(Comp).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

