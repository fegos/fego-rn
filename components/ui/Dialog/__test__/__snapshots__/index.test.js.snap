// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Tests renders correctly 1`] = `
<View
  style={
    Object {
      "zIndex": 1000,
    }
  }
>
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "backgroundColor": "black",
                "opacity": 0.5,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 10,
            "opacity": 1,
            "overflow": "hidden",
            "width": 270,
          }
        }
      >
        <View
          style={
            Object {
              "maxHeight": 1334,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 20,
                  "paddingHorizontal": 22,
                  "paddingTop": 20,
                  "textAlign": "center",
                },
              ]
            }
          >
            标题
          </Text>
          <View
            style={
              Object {
                "paddingBottom": 20,
                "paddingHorizontal": 22,
                "paddingTop": 20,
              }
            }
          >
            <View>
              <Text>
                简单对话框
              </Text>
              Ï
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
